### PR TITLE
Create documentation for connecting Astro to external data services

### PR DESCRIPTION
**Pull request by jwitz**
This document will serve as a hub for administrators that need to connect Astro Clusters to external data sources. We're starting by publishing information about public endpoints and VPC peering, but will expand to document additional connectivity methods and considerations.

### CODE REVIEW

**Review by dnarain-astro — COMMENTED** [review]
Can we do this in stages - lets publish the Public Endpoints and VPC Peering for now. QA will test out transit gateway and privatelink; SRE will work on a runbook for handling customer requests (i.e. what happens after the case is created), but I'd like to get something out there as a stub for later additions.

**Review by dnarain-astro — COMMENTED** [review]
Turns out we don't publish the cluster ID in the UI, so the customer has no way of getting it.  Lets leave it out. If the customer has >1 cluster, support will have have a dialog with the customer to figure out which one needs editing.

Also, "CIDR of the target VPC" should just be VPC CIDR (because we specify it is the target vpc already)

File: astro/connectivity.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+### IP Whitelisting
+
+Some data services, such as Snowflake and Databricks, provide an additional layer of security by requiring you to whitelist a specific IP address before that address can access the service.
+
+Each Cluster has a unique external IP address that will persist for the life of the Cluster. To retrieve the IP for a given Cluster, open a ticket with Astronomer Support and request it. If you have more than one Cluster, you will need to whitelist each Cluster individually on your data service provider.
+
+## VPC Peering
+
+Each Cluster runs in a dedicated VPC. To set up private connectivity between the Astro VPC and your VPC, you can set up a VPC peering connection. Peered VPCs provide additional security by ensuring private connectivity, reduced network transit costs, and simplified network layouts.
```
Comment (jwitz) [inline astro/connectivity.md:34]:
```suggestion
Each Cluster runs in a dedicated VPC. To set up private connectivity between the Astro VPC and a target VPC, you can set up a VPC peering connection. Peered VPCs provide additional security by ensuring private connectivity, reduced network transit costs, and simplified network layouts.
```

File: astro/connectivity.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+### IP Whitelisting
+
+Some data services, such as Snowflake and Databricks, provide an additional layer of security by requiring you to whitelist a specific IP address before that address can access the service.
+
+Each Cluster has a unique external IP address that will persist for the life of the Cluster. To retrieve the IP for a given Cluster, open a ticket with Astronomer Support and request it. If you have more than one Cluster, you will need to whitelist each Cluster individually on your data service provider.
+
+## VPC Peering
+
+Each Cluster runs in a dedicated VPC. To set up private connectivity between the Astro VPC and a target VPC, you can set up a VPC peering connection. Peered VPCs provide additional security by ensuring private connectivity, reduced network transit costs, and simplified network layouts.
```
Comment (jwitz) [inline astro/connectivity.md:34]: What is the typical scope/ context of a VPC connection? Will people be connecting to many VPCs, each its own service, a more general organization-level VPC, or both?

File: astro/connectivity.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- AWS account
+- AWS Region
+- Name of the Cluster
+- VPC CIDR
+
+In response, Astronomer will initiate a peering request. This peering request must be accepted by the owner of the target VPC.
+
+Once peering is set up, continue to work with [Astronomer support](https://support.astronomer.io) and the owner of the target VPC to update the routing tables of both VPCs to direct traffic to each other.
+
+### DNS Considerations with VPC Peering
```
Comment (jwitz) [inline astro/connectivity.md:47]: Is this topic something that all orgs will have to set up for their target VPC? Or is this just a specific way to connect?

**Comment by jwitz** [comment]
@dnarain-astro Sounds good - just updated PR description and text. Left a few comments mostly to better my own understanding, and did a proper pass on formatting, grammar, etc. Let me know what you think!

File: astro/connectivity.md (hunk 2 above)
Comment (dnarain-astro) [reply astro/connectivity.md:32]: Sadly, the answer is both. Most people will have a VPC per service (e.g. a VPC for databricks, a VPC for redshift) and we expect customers to connect to multiple VPCs.

File: astro/connectivity.md (hunk 3 above)
Comment (dnarain-astro) [reply astro/connectivity.md:45]: I don't have a good answer to this one. My expectation is that large customers will use Private Hosted Zones and smaller ones are just using whatever defaults Amazon has in place.  But, we don't have enough usage data to prove or disprove my theory.

**Review by chomipi88 — COMMENTED** [review]
see comments

File: astro/connectivity.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+Before you can orchestrate production-level data in Astro, you first need to make this data accessible to your Astro Clusters. This guide explains how to connect your Astro Clusters to external data sources using the following methods:
+
+- Public Endpoints
+- VPC Peering
+
+## Public Endpoints
+
+The fastest and easiest way to connect Astro to an external data source is by using the data source's publicly accessible API endpoints.
+
+These endpoints can be configured by:
```
[the remark this answers is not in the document]
Comment (jwitz) [reply astro/connectivity.md:21]: @dnarain-astro Probably a question for you

File: astro/connectivity.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+- Setting [Environment Variables](environment-variables.md) with your connection information.
+- Creating an [Airflow Connection](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html).
+
+### IP Whitelisting
+
+Some data services, such as Snowflake and Databricks, provide an additional layer of security by requiring you to whitelist a specific IP address before that address can access the service.
+
+Each Cluster has a unique external IP address that will persist for the life of the Cluster. To retrieve the IP for a given Cluster, open a ticket with [Astronomer support](https://support.astronomer.io) and request it. If you have more than one Cluster, you will need to whitelist each Cluster individually on your data service provider.
```
[the remark this answers is not in the document]
Comment (jwitz) [reply astro/connectivity.md]: Are there functionality considerations given the two IPs? Do you only need to request one per allowlist? cc @dnarain-astro

File: astro/connectivity.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+## VPC Peering
+
+Each Cluster runs in a dedicated VPC. To set up private connectivity between the Astro VPC and a target VPC, you can set up a VPC peering connection. Peered VPCs provide additional security by ensuring private connectivity, reduced network transit costs, and simplified network layouts.
+
+To create a VPC peering connection between a Cluster's VPC and a target VPC, open a support ticket with Astronomer and provide the following information for the target VPC:
+
+- AWS account
+- AWS Region
+- Name of the Cluster
+- VPC CIDR
```
[the remark this answers is not in the document]
Comment (jwitz) [reply astro/connectivity.md:41]: cc @dnarain-astro

File: astro/connectivity.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+## VPC Peering
+
+Each Cluster runs in a dedicated VPC. To set up private connectivity between the Astro VPC and a target VPC, you can set up a VPC peering connection. Peered VPCs provide additional security by ensuring private connectivity, reduced network transit costs, and simplified network layouts.
+
+To create a VPC peering connection between a Cluster's VPC and a target VPC, open a support ticket with Astronomer and provide the following information for the target VPC:
+
+- AWS account
+- AWS Region
+- Name of the Cluster
```
[the remark this answers is not in the document]
Comment (jwitz) [reply astro/connectivity.md]: Which of these details is about your Astro Cluster, and which is about target VPC? @dnarain-astro

**Review by jwitz — COMMENTED** [review]
Thank you for the review @chomipi88 , helped me understand a lot more about the functionality and context with this setup. I tagged @dnarain-astro on questions that I didn't have an answer for, hopefully we can work out all technical details between the 3 of us!

File: astro/connectivity.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (chomipi88) [reply astro/connectivity.md]: No they'd need to whitelist both. Traffic may egress from one or the other depending which node the task is running on.

File: astro/connectivity.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (dnarain-astro) [reply astro/connectivity.md:41]: Its the "about your cluster" verbiage that makes this confusing. HEre's what I recommend:

To create a VPC peering connection between a Cluster's VPC and a target VPC, open a support ticket with Astronomer and provide the following information:
* Name of your Astro Cluster
* AWS Account for the target VPC
* AWS Region for the target VPC
* VPCID of for the target vpc
* CIDR of the target VPC

File: astro/connectivity.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (dnarain-astro) [reply astro/connectivity.md:21]: Yes, we should clarify by saying "The traffic flows directly between the data plane and the API endpoint and is not routed through the control plane"

File: astro/connectivity.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+### IP Allowlist
+
+Some data services, such as Snowflake and Databricks, provide an additional layer of security by requiring you to allowlist a specific IP address before you can access the service.
+
+Each Cluster has a pair of unique external IP address that will persist for the life of the Cluster. These IP addresses are assigned for network address translation, meaning that they're responsible for all outbound traffic from your Cluster to the internet. To retrieve an IP for a given Cluster, open a ticket with [Astronomer support](https://support.astronomer.io) and request it. If you have more than one Cluster, you will need to allowlist each Cluster individually on your data service provider.
+
+## VPC Peering
+
+Each Cluster runs in a dedicated VPC. To set up private connectivity between the Astro VPC and a target VPC, you can set up a VPC peering connection. Peered VPCs provide additional security by ensuring private connectivity, reduced network transit costs, and simplified network layouts.
```
Comment (paolaperaza) [inline astro/connectivity.md:36]:
```suggestion
Each Astro Cluster runs in a dedicated VPC. To set up private connectivity between an Astro VPC and another VPC, you can set up a VPC peering connection. Peered VPCs provide additional security by ensuring private connectivity, reduced network transit costs, and simplified network layouts.
```

File: astro/connectivity.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+To create a VPC peering connection between a Cluster's VPC and a target VPC, open a support ticket with Astronomer and provide the following information:
```
Comment (paolaperaza) [inline astro/connectivity.md:38]:
```suggestion
To create a VPC peering connection between an Astro Cluster's VPC and a target VPC, reach out to [Astronomer support](https://support.astronomer.io) and provide the following information:
```

File: astro/connectivity.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+- Name of your Cluster
+- AWS account for the target VPC
+- AWS Region for the target VPC
+- VPC ID of the target VPC
+- CIDR of the target VPC
```
Comment (paolaperaza) [inline astro/connectivity.md:44]:
```suggestion
- Name of your Astro Cluster
- AWS Account ID for the target VPC
- AWS Region for the target VPC
- VPC ID of the target VPC
- CIDR of the target VPC
```

Assume Account ID here?

File: astro/connectivity.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+In response, Astronomer will initiate a peering request. To connect succesfully, this peering request must be accepted by the owner of the target VPC.
+
+Once peering is set up, continue to work with [Astronomer support](https://support.astronomer.io) and the owner of the target VPC to update the routing tables of both VPCs to direct traffic to each other.
```
Comment (paolaperaza) [inline astro/connectivity.md:48]: I don't feel like we need to link to support here since we already did and this sentence isn't specifically actionable

**Review by paolaperaza — COMMENTED** [review]
A few comments here @jwitz with some clarifying questions for @dnarain-astro. Can you both take a look and then I'll do a final pass? We're close!

File: astro/connectivity.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
+---
+sidebar_label: "Connect to External Services"
+title: "Connect Astro to External Data Services"
+id: connectivity
+description: Learn how to connect Astro to different types of external data sources.
```
[the remark this answers is not in the document]
Comment (jwitz) [reply astro/connectivity.md:5]: This doc might encompass more types of connectivity in the future, I would keep this general.

File: astro/connectivity.md (hunk 8 above)
Comment (jwitz) [reply astro/connectivity.md:36]: If we capitalize "Cluster" I don't think we need to refer to it as "Astro Cluster" beyond the first reference. Added to style guide https://www.notion.so/astronomerio/Astronomer-Industry-Terms-Style-Guide-5d2389d462734e149d2e66a5cf606228

**Comment by jwitz** [comment]
@paolaperaza I think this is now looking good to 🚢 !

File: astro/connect-to-external-services.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
+---
+sidebar_label: "Connect to External Services"
+title: "Connect Astro to External Data Services"
+id: connect-to-external-services
+description: Learn how to connect Astro to different types of external data services.
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:5]:
```suggestion
description: Learn how to connect your Astro Data Plane to different types of external data services.
```

File: astro/connect-to-external-services.md [hunk 14]
```diff
@@ -0,0 +1,56 @@
+---
+sidebar_label: "Connect to External Services"
+title: "Connect Astro to External Data Services"
+id: connect-to-external-services
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:4]:
```suggestion
id: connect-external-services
```

File: astro/connect-to-external-services.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+description: Learn how to connect Astro to different types of external data services.
+---
+
+## Overview
+
+Before you can run pipelines on Astro with real data, you first need to make this data accessible to the Data Plane. This guide explains how to connect your Astro Clusters to external data sources using the following methods:
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:10]:
```suggestion
Before you can run pipelines on Astro with real data, you first need to make this data accessible to the Data Plane. This guide explains how to connect your Astro Clusters to external data services using the following methods:
```

File: astro/connect-to-external-services.md [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+
+- Public Endpoints
+- VPC Peering
+
+If you need to connect to a different type of data source than the ones documented here, reach out to [Astronomer support](https://support.astronomer.io).
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:15]:
```suggestion
If you need to connect to a different type of data service or source than the ones documented here, reach out to [Astronomer support](https://support.astronomer.io).
```

File: astro/connect-to-external-services.md [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+
+## Public Endpoints
+
+The fastest and easiest way to connect Astro to an external data source is by using the data source's publicly accessible endpoints.
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:19]:
```suggestion
The fastest and easiest way to connect Astro to an external data service is by using the data service's publicly accessible endpoints.
```

File: astro/connect-to-external-services.md [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+
+These endpoints can be configured by:
+
+- Setting [Environment Variables](environment-variables.md) with your endpoint information.
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:23]:
```suggestion
- Setting [environment variables](environment-variables.md) on Astro with your endpoint information.
```

File: astro/connect-to-external-services.md [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+- Creating an [Airflow Connection](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html).
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:24]:
```suggestion
- Creating an [Airflow Connection](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html) with your endpoint information.
```

File: astro/connect-to-external-services.md [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+
+Public connection traffic moves directly between your Data Plane and the external data source's API endpoint, meaning that the data in this traffic never reaches the Control Plane, which is managed by Astronomer.
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:26]:
```suggestion
Public connection traffic moves directly between your Data Plane and the external data service's API endpoint. This means that the data in this traffic never reaches the Control Plane, which is managed by Astronomer.
```

File: astro/connect-to-external-services.md [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+
+### IP Allowlist
+
+Some data services, such as Snowflake and Databricks, provide an additional layer of security by requiring you to allowlist a specific IP address before you can access the service.
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:30]:
```suggestion
Some data services, including Snowflake and Databricks, provide an additional layer of security by requiring you to allowlist a specific IP address before you can access the service.
```

File: astro/connect-to-external-services.md [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+
+Each Cluster has a pair of unique external IP address that will persist throughout the lifetime of the Cluster. These IP addresses are assigned for network address translation, meaning that they're responsible for all outbound traffic from your Cluster to the internet. To retrieve an IP for a given Cluster, open a ticket with [Astronomer support](https://support.astronomer.io) and request it. If you have more than one Cluster, you will need to allowlist each Cluster individually on your data service provider.
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:32]:
```suggestion
On Astro, each Cluster has a pair of unique external IP address that will persist throughout the lifetime of the Cluster. These IP addresses are assigned for network address translation, which means that they are responsible for all outbound traffic from your Astro Cluster to the internet. To retrieve an IP for a given Cluster, open a ticket with [Astronomer support](https://support.astronomer.io) and request it. If you have more than one Cluster, you will need to allowlist each Cluster individually on your data service provider.
```

File: astro/connect-to-external-services.md [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+
+## VPC Peering
+
+Each Cluster runs in a dedicated VPC. To set up private connectivity between an Astro VPC and another VPC, you can set up a VPC peering connection. Peered VPCs provide additional security by ensuring private connectivity, reduced network transit costs, and simplified network layouts.
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:36]:
```suggestion
Each Cluster on Astro runs in a dedicated VPC. To set up private connectivity between an Astro VPC and another VPC, you can set up a VPC peering connection. Peered VPCs provide additional security by ensuring private connectivity, reduced network transit costs, and simplified network layouts.
```

File: astro/connect-to-external-services.md [hunk 24]
```diff
@@ -0,0 +1,56 @@
...
+
+To create a VPC peering connection between a Cluster's VPC and a target VPC, reach out to [Astronomer support](https://support.astronomer.io) and provide the following information:
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:38]:
```suggestion
To create a VPC peering connection between an Astro Cluster's VPC and a target VPC, reach out to [Astronomer support](https://support.astronomer.io) and provide the following information:
```

File: astro/connect-to-external-services.md [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+
+- Name of your Cluster
+- AWS Account ID for the target VPC
+- AWS Region for the target VPC
+- VPC ID of the target VPC
+- CIDR of the target VPC
+
+In response, Astronomer will initiate a peering request. To connect successfully, this peering request must be accepted by the owner of the target VPC.
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:46]:
```suggestion
From there, Astronomer will initiate a peering request. To connect successfully, this peering request must be accepted by the owner of the target VPC.
```

File: astro/connect-to-external-services.md [hunk 26]
```diff
@@ -0,0 +1,56 @@
...
+
+Once peering is set up, continue to work with Astronomer Support and the owner of the target VPC to update the routing tables of both VPCs to direct traffic to each other.
```
Comment (paolaperaza) [inline astro/connect-to-external-services.md:48]:
```suggestion
Once peering is set up, the owner of the target VPC can expect to continue to work with our team to update the routing tables of both VPCs to direct traffic to each other.
```

**Review by paolaperaza — APPROVED** [review]
Final copy tweaks, but LGTM!

File: astro/connect-external-services.md [hunk 27]
```diff
@@ -0,0 +1,56 @@
+---
+sidebar_label: "Connect to External Services"
+title: "Connect Astro to External Data Services"
+id: connect-external-services
+description: Learn how to connect your Astro Data Plane to different types of external data services.
+---
+
+## Overview
+
+Before you can run pipelines on Astro with real data, you first need to make your data services accessible to the Data Plane. This guide explains how to securely connect your Astro Clusters to external data services using the following methods:
```
Comment (paolaperaza) [inline astro/connect-external-services.md:10]:
```suggestion
Before you can run pipelines on Astro with real data, you first need to make your data services accessible to Deployments in your Data Plane. This guide explains how to securely connect Astro to external data services using the following methods:
```

File: astro/connect-external-services.md [hunk 28]
```diff
@@ -0,0 +1,56 @@
...
+
+From there, Astronomer will initiate a peering request. To connect successfully, this peering request must be accepted by the owner of the target VPC.
+
+Once peering is set up, the owner of the target VPC can expect to continue to work with our team to update the routing tables of both VPCs to direct traffic to each other.
+
+### DNS Considerations with VPC Peering
+
+To resolve DNS hostnames from your target VPC, your Cluster VPC has **DNS Hostnames**, **DNS Resolutions**, and **Requester DNS Resolution** enabled via AWS [Peering Connection settings](https://docs.aws.amazon.com/vpc/latest/peering/modify-peering-connections.html).  
+
+If your target VPC also resolves DNS hostnames via **DNS Hostnames** and **DNS Resolution**, you must also enable the **Accepter DNS Resolution** setting. This allows the Data Plane to resolve the public DNS hostnames of the target VPC to its private IP addresses. To configure this option, see [AWS Documentation](https://docs.aws.amazon.com/vpc/latest/peering/modify-peering-connections.html).
```
Comment (paolaperaza) [inline astro/connect-external-services.md:54]:
```suggestion
If your target VPC resolves DNS hostnames via **DNS Hostnames** and **DNS Resolution**, you must also enable the **Accepter DNS Resolution** setting. This allows the Data Plane to resolve the public DNS hostnames of the target VPC to its private IP addresses. To configure this option, see [AWS Documentation](https://docs.aws.amazon.com/vpc/latest/peering/modify-peering-connections.html).
```

**Pull request merged**